app.rb

```
@@ -0,0 +1,30 @@
+
+post '/api/test_results' do
+
+  request_body = request.body.read.strip
+
+  # Return error if the body is empty or contains only whitespace
+  if request_body.empty?
+    halt 400, { "error": "No Payload" }.to_json
+  end
+
+  # Parse the JSON and handle parsing errors
+  test_results = JSON.parse(request_body) rescue halt(400, { "error": "Invalid JSON format" }.to_json)
```

rubocop

[convention] Avoid using `rescue` in its modifier form.

(Style/RescueModifier)